Analysis/Tasks/PWGHF/taskBPlus.cxx

```diff
@@ -33,48 +40,205 @@ void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+        registry.fill(HIST("hMassRecSig"), InvMassBplus(candidate), candidate.pt());
+        registry.fill(HIST("hd0Prong0RecSig"), candidate.impactParameter0(), candidate.pt());
+        registry.fill(HIST("hd0Prong1RecSig"), candidate.impactParameter1(), candidate.pt());
+        registry.fill(HIST("hPtProng0RecSig"), candidate.ptProng0(), candidate.pt());
+        registry.fill(HIST("hPtProng1RecSig"), candidate.ptProng1(), candidate.pt());
+        registry.fill(HIST("hImpParProdBPlusRecSig"), candidate.impactParameterProduct(), candidate.pt());
+        registry.fill(HIST("hDeclengthNormRecSig"), candidate.decayLengthXYNormalised(), candidate.pt());
+        registry.fill(HIST("hCPARecSigD0"), D0cand.cpa(), candidate.pt());
+        registry.fill(HIST("hDeclengthRecD0Sig"), D0cand.decayLength(), candidate.pt());
+      } else {
+        registry.fill(HIST("hPtRecBg"), candidate.pt());
+        registry.fill(HIST("hCPARecBg"), candidate.cpa(), candidate.pt());
+        registry.fill(HIST("hCPAxyRecBg"), candidate.cpa(), candidate.pt());
+        registry.fill(HIST("hEtaRecBg"), candidate.eta(), candidate.pt());
+        registry.fill(HIST("hDeclengthRecBg"), candidate.decayLength(), candidate.pt());
+        registry.fill(HIST("hMassRecBg"), InvMassBplus(candidate), candidate.pt());
+        registry.fill(HIST("hd0Prong0RecBg"), candidate.impactParameter0(), candidate.pt());
+        registry.fill(HIST("hd0Prong1RecBg"), candidate.impactParameter1(), candidate.pt());
+        registry.fill(HIST("hPtProng0RecBg"), candidate.ptProng0(), candidate.pt());
+        registry.fill(HIST("hPtProng1RecBg"), candidate.ptProng1(), candidate.pt());
+        registry.fill(HIST("hImpParProdBPlusRecBg"), candidate.impactParameterProduct(), candidate.pt());
+        registry.fill(HIST("hDeclengthNormRecBg"), candidate.decayLengthXYNormalised(), candidate.pt());
+        registry.fill(HIST("hCPARecBgD0"), D0cand.cpa(), candidate.pt());
+        registry.fill(HIST("hDeclengthRecD0Bg"), D0cand.decayLength(), candidate.pt());
+      }
+    } // rec
+
+    // MC gen. level
+    //Printf("MC Particles: %d", particlesMC.size());
+    for (auto& particle : particlesMC) {
+      if (std::abs(particle.flagMCMatchGen()) == 1 << DecayType::BPlusToD0Pi) {
+        if (cutYCandMax >= 0. && std::abs(RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(pdg::Code::kBPlus))) > cutYCandMax) {
+          continue;
         }
+
+        float ptProngs[2], yProngs[2], etaProngs[2];
+        int counter = 0;
+        for (int iD = particle.daughter0(); iD <= particle.daughter1(); ++iD) {
+          ptProngs[counter] = particlesMC.iteratorAt(iD).pt();
+          etaProngs[counter] = particlesMC.iteratorAt(iD).eta();
+
+          auto daught = particlesMC.iteratorAt(iD);
+          yProngs[counter] = RecoDecay::Y(array{daught.px(), daught.py(), daught.pz()}, RecoDecay::getMassPDG(daught.pdgCode()));
+          counter++;
+        }
```

Comment:
* `etaProngs` is not used.
* Avoid calling `particlesMC.iteratorAt(iD)` multiple times. Use `daught` instead.
* The `counter` can go inside `for(...)`.

Reply:
I am not sure I understood the comment "The counter can go inside for(...)".

Reply:
You can initialise and increment `counter` the same way you do it for `iD`.